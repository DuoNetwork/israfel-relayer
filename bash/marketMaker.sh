rm *.log
killall -s KILL node
npm run marketMaker token=aETH env=dev $1 &> aETH.log &