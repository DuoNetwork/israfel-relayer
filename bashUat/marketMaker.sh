rm *.log
killall -s KILL node
npm run marketMaker token=aETH env=uat $1 &> aETH.log &
npm run marketMaker token=sETH env=uat $1 &> sETH.log &
npm run marketMaker token=aETH-M19 env=uat $1 &> aETH-M19.log &
npm run marketMaker token=sETH-M19 env=uat $1 &> sETH-M19.log &