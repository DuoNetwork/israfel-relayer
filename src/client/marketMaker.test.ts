// fix for @ledgerhq/hw-transport-u2f 4.28.0
import '@babel/polyfill';

import marketMaker from './marketMaker';

const obSide = [
	{
		price: 0.001,
		balance: 10,
		count: 1
	},
	{
		price: 0.0015,
		balance: 20,
		count: 2
	},
	{
		price: 0.002,
		balance: 30,
		count: 3
	}
];
test('getSideTotalLiquidity', () => {
	expect(marketMaker.getSideTotalLiquidity(obSide, 1)).toMatchSnapshot();
});
