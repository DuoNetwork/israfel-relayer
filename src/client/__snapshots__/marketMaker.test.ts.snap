// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cancelOrders 1`] = `
Array [
  Array [
    "address",
    "By signing this message, your are confirming to cancel following orders (no gas cost involved): orderHash1,orderHash2",
  ],
]
`;

exports[`cancelOrders 2`] = `
Array [
  Array [
    "aETH|WETH",
    Array [
      "orderHash1",
      "orderHash2",
    ],
    "signature",
  ],
]
`;

exports[`checkAllowance, 0 allowance 1`] = `
Array [
  Array [
    "WETH",
    "address",
  ],
  Array [
    "aETH",
    "address",
  ],
  Array [
    "bETH",
    "address",
  ],
  Array [
    "WETH",
    "address",
    "custodianAddr",
  ],
]
`;

exports[`checkAllowance, 0 allowance 2`] = `
Array [
  Array [
    "WETH",
    "address",
  ],
  Array [
    "aETH",
    "address",
  ],
  Array [
    "bETH",
    "address",
  ],
  Array [
    "WETH",
    "address",
    "custodianAddr",
  ],
]
`;

exports[`checkAllowance, 0 allowance 3`] = `
Array [
  Array [
    "WETH|
	address|undefined",
  ],
  Array [
    "aETH|
	address|undefined",
  ],
  Array [
    "bETH|
	address|undefined",
  ],
  Array [
    "WETH|
	address|custodianAddr",
  ],
]
`;

exports[`checkAllowance, already approved 1`] = `
Array [
  Array [
    "WETH",
    "address",
  ],
  Array [
    "aETH",
    "address",
  ],
  Array [
    "bETH",
    "address",
  ],
  Array [
    "WETH",
    "address",
    "custodianAddr",
  ],
]
`;

exports[`connectToRelayer 1`] = `
Array [
  Array [
    [Function],
  ],
]
`;

exports[`connectToRelayer 2`] = `
Array [
  Array [
    [Function],
    [Function],
    [Function],
  ],
]
`;

exports[`connectToRelayer 3`] = `
Array [
  Array [
    [Function],
    [Function],
  ],
]
`;

exports[`connectToRelayer 4`] = `
Array [
  Array [
    [Function],
    [Function],
  ],
]
`;

exports[`connectToRelayer 5`] = `
Array [
  Array [
    [Function],
    3600000,
  ],
]
`;

exports[`createOrderBookFromNav 1`] = `
Array [
  Array [
    Object {},
    "aETH|WETH",
    0.0099,
    true,
  ],
  Array [
    Object {},
    "aETH|WETH",
    0.0101,
    false,
  ],
  Array [
    Object {},
    "bETH|WETH",
    0.0119,
    true,
  ],
  Array [
    Object {},
    "bETH|WETH",
    0.0121,
    false,
  ],
]
`;

exports[`createOrderBookSide 1`] = `
Array [
  Array [
    "address",
    "aETH|WETH",
    0.0001,
    22.5,
    true,
    1234567890000,
  ],
  Array [
    "address",
    "aETH|WETH",
    0,
    22.5,
    true,
    1234567890000,
  ],
  Array [
    "address",
    "aETH|WETH",
    -0.0001,
    22.5,
    true,
    1234567890000,
  ],
  Array [
    "address",
    "aETH|WETH",
    -0.0002,
    22.5,
    true,
    1234567890000,
  ],
]
`;

exports[`getEthPrice 1`] = `150`;

exports[`getEthPrice, no ETH price 1`] = `0`;

exports[`handleOrderBookUpdate 1`] = `
Array [
  Array [
    Object {},
    Object {},
    "aETH|WETH",
  ],
]
`;

exports[`handleOrderHistory 1`] = `
Array [
  99.9868,
  86,
  84,
]
`;

exports[`handleOrderHistory 2`] = `
Array [
  "aETH|WETH",
  Array [
    "orderHash3",
  ],
]
`;

exports[`handleOrderHistory 3`] = `
Array [
  "bETH|WETH",
  Array [
    "orderHash2",
    "orderHash4",
  ],
]
`;

exports[`handleOrderHistory 4`] = `
Array [
  Array [
    "aETH|WETH",
  ],
  Array [
    "bETH|WETH",
  ],
]
`;

exports[`handleUserOrder, add, ask 1`] = `
Array [
  6,
  190,
  200,
]
`;

exports[`handleUserOrder, add, ask 2`] = `
Array [
  Array [
    Object {},
    Object {},
    "aETH|WETH",
  ],
]
`;

exports[`handleUserOrder, add, bid 1`] = `
Array [
  5.99,
  200,
  200,
]
`;

exports[`handleUserOrder, add, bid 2`] = `
Array [
  Array [
    Object {},
    Object {},
    "aETH|WETH",
  ],
]
`;

exports[`handleUserOrder, add, cancel too far away 1`] = `
Array [
  5.99,
  200,
  200,
]
`;

exports[`handleUserOrder, add, cancel too far away 2`] = `
Array [
  Array [
    Object {},
    Object {},
    "aETH|WETH",
  ],
]
`;

exports[`handleUserOrder, add, cancel too far away 3`] = `
Array [
  "aETH|WETH",
  Array [
    "orderHash1",
  ],
]
`;

exports[`handleUserOrder, terminate, ask 1`] = `
Array [
  6,
  210,
  200,
]
`;

exports[`handleUserOrder, terminate, ask 2`] = `
Array [
  Array [
    Object {},
    Object {},
    "aETH|WETH",
  ],
]
`;

exports[`handleUserOrder, terminate, bid 1`] = `
Array [
  6.01,
  200,
  200,
]
`;

exports[`handleUserOrder, terminate, bid 2`] = `
Array [
  Array [
    Object {},
    Object {},
    "aETH|WETH",
  ],
]
`;

exports[`handleUserOrder, terminate, bid bToken 1`] = `
Array [
  6.01,
  200,
  200,
]
`;

exports[`handleUserOrder, terminate, bid bToken 2`] = `
Array [
  Array [
    Object {},
    Object {},
    "bETH|WETH",
  ],
]
`;

exports[`handleUserOrder, update, ask 1`] = `
Array [
  6,
  240,
  200,
]
`;

exports[`handleUserOrder, update, ask 2`] = `
Array [
  Array [
    Object {},
    Object {},
    "aETH|WETH",
  ],
]
`;

exports[`handleUserOrder, update, bid 1`] = `
Array [
  6.04,
  200,
  200,
]
`;

exports[`handleUserOrder, update, bid 2`] = `
Array [
  Array [
    Object {},
    Object {},
    "aETH|WETH",
  ],
]
`;

exports[`initialize 1`] = `
Array [
  Array [
    "aETH",
  ],
]
`;

exports[`initialize 2`] = `
Array [
  Array [
    "WETH",
    "address",
  ],
  Array [
    "aETH",
    "address",
  ],
  Array [
    "bETH",
    "address",
  ],
]
`;

exports[`initialize 3`] = `
Array [
  Array [
    "address",
  ],
]
`;

exports[`initialize, no a token 1`] = `[Error: no aToken]`;

exports[`initialize, no b token 1`] = `[Error: no bToken]`;

exports[`maintainBalance, short of token 1`] = `
Array [
  7.114996114996115,
  300,
  300,
]
`;

exports[`maintainBalance, short of token 2`] = `
Array [
  Array [
    "address",
    "privateKey",
    9000000000,
    160000,
    3.88500389,
    "wethAddr",
  ],
]
`;

exports[`maintainBalance, short of weth 1`] = `
Array [
  6,
  200,
  200,
]
`;

exports[`maintainBalance, short of weth 2`] = `
Array [
  Array [
    "WETH",
    "0x00D8d0660b243452fC2f996A892D3083A903576F",
    "address",
    "address",
    5,
  ],
]
`;

exports[`maintainBalance, surplus of token 1`] = `
Array [
  5.04615385,
  300,
  300,
]
`;

exports[`maintainBalance, surplus of token 2`] = `
Array [
  Array [
    "address",
    "privateKey",
    200,
    200,
    9000000000,
    80000,
  ],
]
`;

exports[`maintainBalance, surplus of token 3`] = `
Array [
  Array [
    3.04615385,
    "address",
  ],
]
`;

exports[`maintainBalance, surplus of weth 1`] = `
Array [
  6,
  200,
  200,
]
`;

exports[`maintainBalance, surplus of weth 2`] = `
Array [
  Array [
    "WETH",
    "address",
    "0x00D8d0660b243452fC2f996A892D3083A903576F",
    "address",
    6,
  ],
]
`;

exports[`makeOrders, arbitrage occurs, take asks 1`] = `
Array [
  "bETH|WETH",
  false,
  Array [
    Object {
      "balance": 20,
      "count": 1,
      "price": 0.0085,
    },
    Object {
      "balance": 20,
      "count": 1,
      "price": 0.0084,
    },
    Object {
      "balance": 20,
      "count": 1,
      "price": 0.0083,
    },
    Object {
      "balance": 20,
      "count": 1,
      "price": 0.0082,
    },
  ],
]
`;

exports[`makeOrders, arbitrage occurs, take asks 2`] = `
Array [
  "bETH|WETH",
  Array [
    "orderHash4",
  ],
]
`;

exports[`makeOrders, arbitrage occurs, take bids 1`] = `
Array [
  "bETH|WETH",
  true,
  Array [
    Object {
      "balance": 20,
      "count": 1,
      "price": 0.0091,
    },
    Object {
      "balance": 20,
      "count": 1,
      "price": 0.009,
    },
  ],
]
`;

exports[`makeOrders, create ask order 1`] = `
Array [
  "aETH|WETH",
  0.006705,
  false,
  1,
]
`;

exports[`makeOrders, create bid order 1`] = `
Array [
  "aETH|WETH",
  0.006305,
  true,
  1,
]
`;

exports[`takeOneSideOrders 1`] = `
Array [
  Array [
    "address",
    "aETH|WETH",
    0.001,
    20,
    false,
    1234567890000,
  ],
  Array [
    "address",
    "aETH|WETH",
    0.0014,
    20,
    false,
    1234567890000,
  ],
]
`;
